### PR TITLE
Add `msqDenySelect` planner config to (dis)allow MSQ SELECT queries.

### PR DESCRIPTION
**Pull request by abhishekrb19**
### Description

This PR adds a new planner configuration `msqDenySelect` that, when set to true,  MSQ `SELECT` statements will be blocked from planning. This parameter is set to `false` by default, so _all_ queries will plan.

This PR has:

- [x] been self-reviewed.
- [x] added documentation for new or modified features or behaviors.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.

### CODE REVIEW

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java [hunk 1]
```diff
@@ -183,4 +190,53 @@ public void testArrayAggQueryOnComplexDatatypes()
       );
     }
   }
+
+  @Test
+  public void testMsqDenySelectEnabledQuery()
+  {
+    msqCompatible();
```
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java:197]: These tests probably want to be in `CalciteInsertDmlTest`, if they can share config. Those tests are run only for MSQ. The test in this form runs for both MSQ and non-MSQ, which may not be what you wanted.

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java [hunk 2]
```diff
@@ -183,4 +190,53 @@ public void testArrayAggQueryOnComplexDatatypes()
...
+    }
+    catch (SqlPlanningException e) {
+      Assert.assertTrue(
+          e.getMessage().contains("Cannot execute SELECT with SQL engine 'msq-task'")
+      );
+    }
+  }
+
+  @Test
+  public void testMsqDenySelectDisabledQuery()
```
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryMSQTest.java:218]: A test that denies access should probably expect an error, I would think.

File: processing/src/main/java/org/apache/druid/query/QueryContexts.java [hunk 3]
```diff
@@ -110,6 +110,7 @@
   public static final boolean DEFAULT_ENABLE_DEBUG = false;
   public static final int DEFAULT_IN_SUB_QUERY_THRESHOLD = Integer.MAX_VALUE;
   public static final boolean DEFAULT_ENABLE_TIME_BOUNDARY_PLANNING = false;
+  public static final boolean DEFAULT_MSQ_DENY_SELECT = false;
```
Comment (paul-rogers) [inline processing/src/main/java/org/apache/druid/query/QueryContexts.java:113]: No longer needed.

**Comment by FrankChen021** [comment]
> It looks like only 1 GHA check ran. The tests didn't seem to run

This's because the ASF recently requires committers to click a 'apporve' button to run CI. We discussed this matter in dev mailing thread and wanted to change it back.

**Comment by abhishekrb19** [comment]
Closing this in favor of a different approach which would include query kind in the `EXPLAIN PLAN` output.

**Pull request closed without merging**